### PR TITLE
feat: split AsideHeader to compound components

### PR DESCRIPTION
**Pull request by ldrv565**
according to this PR we do not expect any functional changes, only adding the ability to use AsideHeader in a different way

### CODE REVIEW

**Comment by DarkGenius** [comment]
Please add to README example of use PageLayout

File: README.md [hunk 1]
```diff
@@ -24,6 +24,36 @@ npm install --dev @gravity-ui/uikit@^3.0.2 @bem-react/classname@1.6.0 react@^16.
...
+## Optimization
+If your app content needs to be rendered faster than by passing it throw `AsideHeader` props,
+you may need to switch usage of `AsideHeader` to advanced style with `PageLayout` like this:
+
+```diff
+-import {AsideHeader} from '@gravity-ui/navigation';
++import {PageLayout} from '@gravity-ui/navigation';
++
++const PageLayoutAside = React.lazy(
++    import('@gravity-ui/navigation').then((module) => ({default: module.PageLayoutAside})),
```
Comment (ldrv565) [inline README.md:39]: here may be trouble with forwarding ref, I will think about it tomorrow

File: package.json [hunk 2]
```diff
@@ -81,6 +80,7 @@
     "rollup-plugin-json": "^4.0.0",
     "rollup-plugin-peer-deps-external": "^2.2.4",
     "rollup-plugin-postcss": "^4.0.2",
+    "rollup-plugin-typescript2": "^0.36.0",
```
Comment (DarkGenius) [inline package.json:83]: Reason for using this package?

Comment (ldrv565) [reply package.json:83]: @rollup/plugin-typescript supports only one outDir

Comment (DarkGenius) [reply package.json:83]: We didn't plan to change the plugin for typescript here. The current plugin is considered more stable and faster

Comment (ldrv565) [reply package.json:83]: with current plugin we have to duplicate the same build with different outDir and than build time will be two times greater

Comment (ldrv565) [reply package.json:83]: seems like it was fixed in 11 version, I will just update @rollup/plugin-typescript

**Review by DarkGenius — CHANGES_REQUESTED** [review]
`PageLayout` and `PageLayoutAside` components are directly related to `AsideHeader`. Let's move them to `AsideHeader/components`.

**Review by DarkGenius — APPROVED** [review]

**Pull request merged**